### ISSUE TITLE
tweak `atexit` to get rid of malloc

### ISSUE THREAD

**Issue by fenugrec** [candle-usb/candleLight_fw#130]
_Originally posted by @fenugrec in https://github.com/candle-usb/candleLight_fw/issues/128#issuecomment-1311092861_

See comment above;
in certain cases the builtint atexit implementation pulls in malloc and some other stuff. 

More information at https://stackoverflow.com/questions/48711221/how-to-prevent-inclusion-of-c-library-destructors-and-atexit

**Comment by fenugrec** [comment]
`-fno-use-cxa-atexit` will not work :
 ` command-line option '-fno-use-cxa-atexit' is valid for C++/ObjC++ but not for C [-Werror]`

But, overriding the weak symbol `__register_exitproc()` seems to work, saving almost 300 bytes:

```
Before: 

   text    data     bss     dec     hex filename
  14296     188    3576   18060    468c cannette_fw 

With override:

   text    data     bss     dec     hex filename
  13912     184    3432   17528    4478 cannette_fw 

```

**Comment by fenugrec** [comment]
closed via #147
